opentelemetry-exporter-geneva/geneva-uploader/src/config_service/client.rs

```diff
@@ -0,0 +1,552 @@
+/// * `namespace` - Namespace for the configuration
+/// * `region` - Azure region (e.g., "westus2")
+/// * `config_major_version` - Major version of the configuration schema
+/// * `auth_method` - Authentication method to use (Certificate or ManagedIdentity)
+///
+/// # Example
+/// ```ignore
+/// let config = GenevaConfigClientConfig {
+///     endpoint: "https://example.geneva.com".to_string(),
+///     environment: "prod".to_string(),
+///     account: "myaccount".to_string(),
+///     namespace: "myservice".to_string(),
+///     region: "westus2".to_string(),
+///     config_major_version: 1,
+///     auth_method: AuthMethod::Certificate {
+///         path: "/path/to/cert.p12".to_string(),
+///         password: "password".to_string(),
+///     },
+/// };
+/// ```
+#[allow(dead_code)]
+#[derive(Clone)]
+pub(crate) struct GenevaConfigClientConfig {
+    pub(crate) endpoint: String,
+    pub(crate) environment: String,
+    pub(crate) account: String,
+    pub(crate) namespace: String,
+    pub(crate) region: String,
+    pub(crate) config_major_version: u32,
+    pub(crate) auth_method: AuthMethod, // agent_identity and agent_version are hardcoded for now
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Deserialize)]
+pub(crate) struct IngestionGatewayInfo {
+    #[serde(rename = "Endpoint")]
+    pub(crate) endpoint: String,
+    #[serde(rename = "AuthToken")]
+    pub(crate) auth_token: String,
+    #[serde(rename = "AuthTokenExpiryTime")]
+    pub(crate) auth_token_expiry_time: String,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub(crate) struct MonikerInfo {
+    pub name: String,
+    pub account_group: String,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+struct StorageAccountKey {
+    #[serde(rename = "AccountMonikerName")]
+    account_moniker_name: String,
+    #[serde(rename = "AccountGroupName")]
+    account_group_name: String,
+    #[serde(rename = "IsPrimaryMoniker")]
+    is_primary_moniker: bool,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+struct GenevaResponse {
+    #[serde(rename = "IngestionGatewayInfo")]
+    ingestion_gateway_info: IngestionGatewayInfo,
+    // TODO: Make storage_account_keys optional since it might not be present in all responses
+    #[serde(rename = "StorageAccountKeys", default)]
+    storage_account_keys: Vec<StorageAccountKey>,
+    // Keep tag_id as it might be used for validation
+    #[serde(rename = "TagId")]
+    tag_id: String,
+}
+
+#[allow(dead_code)]
+struct CachedAuthData {
+    // Store the complete token and moniker info
+    auth_info: (IngestionGatewayInfo, MonikerInfo),
+    // Store the endpoint from token for quick access
+    token_endpoint: String,
+    // Store expiry separately for quick access
+    token_expiry: DateTime<Utc>,
+}
+
+#[allow(dead_code)]
+pub(crate) struct GenevaConfigClient {
+    config: GenevaConfigClientConfig,
+    http_client: Client,
+    cached_data: RwLock<Option<CachedAuthData>>,
```

Comment:
nit: Add a TODO here to revisit this and check if we can get rid of the `RwLock`.